airflow/utils/sqlalchemy.py

```diff
@@ -42,21 +43,25 @@
 
 class UtcDateTime(TypeDecorator):
     """
-    Almost equivalent to :class:`~sqlalchemy.types.DateTime` with
+    Almost equivalent to :class:`~sqlalchemy.types.TIMESTAMP` with
     ``timezone=True`` option, but it differs from that by:
 
     - Never silently take naive :class:`~datetime.datetime`, instead it
       always raise :exc:`ValueError` unless time zone aware value.
     - :class:`~datetime.datetime` value's :attr:`~datetime.datetime.tzinfo`
       is always converted to UTC.
-    - Unlike SQLAlchemy's built-in :class:`~sqlalchemy.types.DateTime`,
+    - Unlike SQLAlchemy's built-in :class:`~sqlalchemy.types.TIMESTAMP`,
       it never return naive :class:`~datetime.datetime`, but time zone
       aware value, even with SQLite or MySQL.
-    - Always returns DateTime in UTC
+    - Always returns TIMESTAMP in UTC
 
     """
 
-    impl = DateTime(timezone=True)
+    impl = (
+        TIMESTAMP(timezone=True)
+        .with_variant(mssql.DATETIME2(precision=6), 'mssql')
+        .with_variant(mysql.TIMESTAMP(fsp=6), 'mysql')
+    )
```

Comment:
`TIMESTAMP` is a subclass of `DateTime` with fractional seconds support(https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.DateTime). 
We use it in migration file(`db_types.TIMESTAMP`). This change makes it the same with DB such that the default is `TIMESTAMP` with the different variations. This will help in other DBs not supported yet to be in uniform as we would use `db_types.TIMESTAMP` for migrations involving datetime+timezone.